internal/agent/upgrade.go

```diff
@@ -65,11 +67,14 @@ func Upgrade(
 		return err
 	}
 
-	upgradeAction := action.NewUpgradeAction(c, upgradeSpec)
-
-	err = upgradeAction.Run()
-	if err != nil {
-		return err
+	if boardutils.GetAndroidBoardModel() == constants.QCS6490 {
```

Comment:
what if we call it GenericAndroidModel for now? is there anything really specific of QCS?

Reply:
I have zero idea, that's why I made it very specific for this board as others may not work the same